dxp/resource_injection.md

```diff
@@ -0,0 +1,106 @@
+# DXP Support for Injection of Dynamic Resources
+
+This document describes different APIs that can be used with DXP to inject resources (`.js`, `.css`, ...) to a HTML page, thus letting the user extend currently existing artifacts without any need to overwrite them (like it was previously done with hooks in older Liferay versions).
+
+# `DynamicInclude` in `portal-kernel`
+
+[`DynamicInclude`](https://github.com/liferay/liferay-portal/blob/973e6199844436c22d237e2daf4821d0f9af5362/portal-kernel/src/com/liferay/portal/kernel/servlet/taglib/DynamicInclude.java) is the lowest level API that lets programmers inject arbitrary content in any rendered page through access to its [`HttpServletResponse`](https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html).
+
+`DynamicInclude`s attach to a unique `key` which identifies where the content is inserted. An example of such key can be `/html/common/themes/top_head.jsp#post` which makes the content to be inserted in [this DXP's core JSP](https://github.com/liferay/liferay-portal/blob/e3ffac158e0ec5acc5c67069fbd7ba688d3c78d4/portal-web/docroot/html/common/themes/top_head.jsp#L216).
```

Comment:
"to be inserted" → "be inserted"

I don't really get what "this DXP's core JSP"...

Reply:
Changed "this DXP's core JSP" to "this JSP file"